src/IO.Ably.Shared/TokenDetails.cs

```diff
@@ -155,21 +157,23 @@ public static class TokenDetailsExtensions
         /// Checks whether the token is valid.
         /// </summary>
         /// <param name="token"><see cref="TokenDetails"/>.</param>
-        /// <param name="now">the correct instance of now to compare with the token.</param>
+        /// <param name="serverTime">the server time instance of now to compare with the token.</param>
         /// <returns>true / false.</returns>
-        public static bool IsValidToken(this TokenDetails token, DateTimeOffset now)
+        public static bool IsValidToken(this TokenDetails token, DateTimeOffset? serverTime)
         {
             if (token == null)
             {
                 return false;
             }
 
-            if (token.Expires == DateTimeOffset.MinValue)
+            // (RSA4b1) We can only check validity of the token if it is a full TokenDetails object
+            // and we already have the server time
+            if (serverTime is null || token.CanBeUsedToCheckExpiry == false)
             {
                 return true;
             }
 
-            return !token.IsExpired(now);
+            return token.IsExpired(serverTime.Value) == false;
```

Comment:
Not sure why the explicit `== false` here, as I've commented on elsewhere. Why not?:

```suggestion
            return !token.IsExpired(serverTime.Value);
```